### ISSUE TITLE
Deleting an observation crashes the app

### ISSUE THREAD

**Issue by jtklein** [inaturalist/iNaturalistReactNative#545]
1. On the observation list
2. Navigate to one observation
3. ... button, delete, confirm
4. App navigates back to observation list and after few moments crashes.

With

```
ERROR : JS Error: Fatal: Error: Exception in HostFunction: Accessing object of type Observation which has been invalidated or deleted
    at get__synced_at (native)
    at wasSynced (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:112413:20)
    at ObsUploadStatus (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:322240:42)
    at renderWithHooks (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:9806:33)
    at updateFunctionComponent
```

and

```
Possible Unhandled Promise Rejection (id: 0):
Error: Exception in HostFunction: Cannot access realm that has been closed.
Error: Exception in HostFunction: Cannot access realm that has been closed.
    at objects (native)
    at anonymous (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:112638:53)
    at anonymous (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:112643:57)
    at anonymous (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:112467:54)
    at filter (native)
    at upsertRemoteObservations (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:112466:48)
    at _callee20$ (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.inaturalist.iNatMobileBeta:371049:62)
    at call (native)
    at tryCatch 
```

**Comment by Chrischuck** [comment]
I can take a look at this

**Comment by kueda** [comment]
This bug is back in 0.4.0 (44). This could benefit from an integration test.

**Comment by kueda** [comment]
FWIW, I worked on this earlier this week, but only got as far as realizing that an integration test is impossible because the React Navigation header uses native rendering, and an e2e test was stalling and matchers weren't working when returning to My Observations. The e2e problems may have been due to location fetching not working in the e2e simulators, which may have been fixed by https://github.com/inaturalist/iNaturalistReactNative/pull/643, so that's probably worth revisiting. I expect the underlying problem has to do with the observations rendered in My Observations not getting updated when an observation is deleted.
